devicetypes/smartthings/zwave-lock.src/zwave-lock.groovy

```diff
@@ -768,14 +768,14 @@ def zwaveEvent(UserCodeReport cmd) {
 		} else {
 			// We'll land here during scanning of codes
 			codeName = getCodeName(lockCodes, codeID)
+			def changeType = getChangeType(lockCodes, codeID)
 			if (!lockCodes[codeID]) {
-				map.value = "$codeID set"
```

Comment:
My apologies for being a bit slow to pick up what's going on here, but why are we removing this value from these events?

Also what's the difference between the two events created in this case? (one from createSetEvent and one from the map) Are they each used differently?

Reply:
Hi Steven. We are not removing the value from events. The value will be added at line number 777. We have just shifted it down because the logic is common for both the `if` and `else` loops.

- `map` will create an event with name `codeChanged`. This event is handled by Smart Locks SmartApp in case of code creation, code deletion, code update etc.
- `codeSetEvent` will create two events:
   1. An event that updates `lockCodes` attribute.
   2. `codeReport` event. This event is legacy event and is needed for supporting existing SmartApps of 3rd party developers.